tests/components/dsmr/test_mbus_migration.py

```diff
@@ -302,6 +325,23 @@ async def test_migrate_gas_with_devid_to_mbus(
     # after receiving telegram entities need to have the chance to be created
     await hass.async_block_till_done()
 
+    # Check a new device is not created and the old device has not been removed
+    assert len(device_registry.devices) == 1
+    assert device_registry.async_get(device.id)
+    new_entity = entity_registry.async_get("sensor.gas_meter_reading")
+    new_device = device_registry.async_get(new_entity.device_id)
+    assert new_device.id == device.id
+    new_dev_entities = er.async_entries_for_device(
+        entity_registry, new_device.id, include_disabled_entities=True
+    )
+    assert new_dev_entities == [new_entity]
+
+    # Check entities are still connected to the old device
+    dev_entities = er.async_entries_for_device(
+        entity_registry, device.id, include_disabled_entities=True
+    )
+    assert dev_entities == [new_entity]
+
```

Comment:
```suggestion
```

Since we do `assert new_device.id == device.id` on L333, this is doing the same check as the one above, right?

Reply:
Yes, absolutely! Fixed.